### PR TITLE
NIFI-1563: Federate requests and merge responses from nodes instead o…

### PR DESCRIPTION
**Pull request by markap14**
Refactoring for the 1.0 baseline that eliminates storage of nodes' stats and bulletins, etc. from NCM and instead replicates the requests to nodes and merges responses inline

### CODE REVIEW

**Comment by mcgilman** [comment]
Reviewing...

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-client-dto/pom.xml [hunk 1]
```diff
@@ -26,5 +26,13 @@
...
             <artifactId>swagger-annotations</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-utils</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-api</artifactId>
+        </dependency>
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-client-dto/pom.xml]: Are these dependencies necessary? Would like to have this artifact as 'light' and 'portability' as possible. Should just be definitions of transport objects for using the REST API.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/CounterDTO.java [hunk 2]
```diff
@@ -98,4 +98,12 @@ public void setValueCount(Long valueCount) {
         this.valueCount = valueCount;
     }
 
+    @Override
+    public CounterDTO clone() {
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/CounterDTO.java]: Was this the intended implementation of this method? This implementation doesn't seem consistent with others.

Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/CounterDTO.java]: @markap14 There appears to be about 5 instances of super.clone(). Given the ambiguous specification on the implementation of Object.clone and the inconsistency of doing a manual clone explicitly elsewhere can we update these?

**Comment by mcgilman** [comment]
@markap14 Unfortunately, the changeset is too big for me to comment on directly. But in nf-counters-table.js the Ajax request on line 281 does not need to set the nodewise flag to true. We do not currently provide a node by node breakdown of the Counters. When we do offer that, we should add that flag back in.

**Comment by mcgilman** [comment]
Also, in nf-cluster-search.js we can delete line 72 and 172. We no longer need to set:

`nf.SummaryTable.systemDiagnosticsUrl`

Setting the clusterNodeId on the preceding line should be sufficient.

**Comment by mcgilman** [comment]
@markap14 Proposed commit for my comments [1]. If this looks good to you, I'll include them in your PR.

[1] https://github.com/mcgilman/nifi/commit/9796e7620cb064653293d8bc0b2293b8b063f3b7

**Comment by markap14** [comment]
@mcgilman I looked at the commit that you proposed. Looks good! +1. Thanks for jumping in there and fixing that!

**Pull request closed without merging**